sdk/core/Azure.Core/src/Pipeline/HttpWebRequestTransport.cs

```diff
@@ -0,0 +1,302 @@
+
+        /// <summary>
+        /// Creates a new instance of <see cref="HttpWebRequestTransport"/>
+        /// </summary>
+        public HttpWebRequestTransport()
+        {
+            if (HttpEnvironmentProxy.TryCreate(out IWebProxy webProxy))
+            {
+                _proxy = webProxy;
+            }
+        }
+
+        /// <inheritdoc />
+        public override void Process(HttpMessage message)
+        {
+            ProcessInternal(message, false).EnsureCompleted();
+        }
+
+        /// <inheritdoc />
+        public override async ValueTask ProcessAsync(HttpMessage message)
+        {
+            await ProcessInternal(message, true);
+        }
+
+        private async ValueTask ProcessInternal(HttpMessage message, bool async)
+        {
+            var request = CreateRequest(message.Request);
+            using var registration = message.CancellationToken.Register(state => ((HttpWebRequest) state).Abort(), request);
+            try
+            {
+                if (message.Request.Content != null)
+                {
+                    if (request.ContentLength == -1 &&
+                        message.Request.Content.TryComputeLength(out var length))
+                    {
+                        request.ContentLength = length;
+                    }
+
+                    var requestStream = async ? await request.GetRequestStreamAsync().ConfigureAwait(false) : request.GetRequestStream();
+
+                    if (async)
+                    {
+                        await message.Request.Content.WriteToAsync(requestStream, message.CancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        message.Request.Content.WriteTo(requestStream, message.CancellationToken);
+                    }
+                }
+
+                var webResponse = async ? await request.GetResponseAsync().ConfigureAwait(false) : request.GetResponse();
+
+                message.Response = new HttpWebResponseImplementation(message.Request.ClientRequestId, (HttpWebResponse) webResponse);
+            }
+            // WebException is thrown in the case of .Abort() call
+            catch (WebException) when (message.CancellationToken.IsCancellationRequested)
+            {
+                message.CancellationToken.ThrowIfCancellationRequested();
+            }
+            catch (WebException webException)
+            {
+                throw new RequestFailedException(0, webException.Message);
+            }
+        }
+
+        private HttpWebRequest CreateRequest(Request messageRequest)
+        {
+            var request = WebRequest.CreateHttp(messageRequest.Uri.ToUri());
+            request.Method = messageRequest.Method.Method;
+            request.Proxy = _proxy;
+            foreach (var messageRequestHeader in messageRequest.Headers)
+            {
+                if (messageRequestHeader.Name == HttpHeader.Names.ContentLength)
```

Comment:
HTTP headers are case-insensitive. May want to do a `OrdinalIgnoreCase` comparison on these (below as well).

Reply:
Great point.